presto-kafka/src/main/java/io/prestosql/plugin/kafka/KafkaProducerFactory.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kafka;
+
+import org.apache.kafka.clients.producer.KafkaProducer;
+
+import java.util.Properties;
+
+public interface KafkaProducerFactory
+{
+    default KafkaProducer<Object, Object> create(String keySerializer, String messageSerializer)
+    {
+        return new KafkaProducer<>(configure(keySerializer, messageSerializer));
+    }
+
+    Properties configure(String keySerializer, String messageSerializer);
```

Comment:
Should it be part of interface?

Since we have 1 implementation, do we need the interface at all?

Reply:
I don't think we need the interface unless we want to create producers with different properties. All I can think of for this is maybe different producers for different size inserts? Like if it's a very large insert we could change the batch size or buffer?

Reply:
There is no way to pass `Properties` _in_ (and no way to know what valid prop keys there are).
You seem to have the method as a helper to the default method. A regular class, and a protected methodn would be more suitable.